File: baekjoon/BinarySearch/1920_수 찾기.py
import sys
from tabnanny import check
input = sys.stdin.readline
n = int(input())
nArr = list(map(int, input().split(" ")))
nArr.sort()
m = int(input())
mArr = list(map(int, input().split(" ")))


def solution():
    for find in mArr:
        start = 0
        end = n-1
        check = False
        while start <= end:
            mid = (start+end) // 2
            if nArr[mid] == find:
                check = True
                break
            elif nArr[mid] > find:
                end = mid - 1
            else:
                start = mid + 1
        print(1 if check else 0)


solution()

File: baekjoon/BinarySearch/1920_수 찾기.js
const fs = require('fs')
let [n, nArr, m, mArr] = fs
  .readFileSync('../input.txt')
  .toString()
  .trim()
  .split('\n')
n = Number(n)
nArr = nArr
  .split(' ')
  .map(Number)
  .sort((a, b) => a - b)
mArr = mArr.split(' ').map(Number)

function solution() {
  const answer = []
  for (const find of mArr) {
    let start = 0
    let end = n - 1
    let check = false

    while (start <= end) {
      let mid = parseInt((start + end) / 2)
      if (nArr[mid] === find) {
        check = true
        break
      } else if (nArr[mid] > find) {
        end = mid - 1
      } else {
        start = mid + 1
      }
    }
    answer.push(check ? 1 : 0)
  }
  // 마지막에 한 번에 출력시 시간 초과 회피 가능
  console.log(answer.join('\n'))
}

solution()
